src/pagination/pagination.component.ts

```diff
@@ -106,10 +105,24 @@ import { ExperimentalService } from "./../experimental.module";
 					'bx--select__page-number' : isExperimental
 				}">
 					<label [for]="currentPageSelectId" class="bx--label bx--visually-hidden">{{itemsPerPageText | async}}</label>
-					<select [id]="currentPageSelectId" class="bx--select-input" [(ngModel)]="currentPage">
-						<option *ngFor="let i of range(lastPage + 1, 1)" class="bx--select-option" [value]="i">{{i}}</option>
+					<input
+						*ngIf="pageOptions.length > pageSelectThreshold"
+						style="padding-right: 1rem; margin-right: 1rem;"
+						[id]="currentPageSelectId"
+						type="number"
+						min="1"
+						[max]="pageOptions.length"
+						class="bx--select-input"
+						[(ngModel)]="currentPage">
+					<select
+						*ngIf="pageOptions.length <= pageSelectThreshold"
+						[id]="currentPageSelectId"
+						class="bx--select-input"
+						[(ngModel)]="currentPage">
+						<option *ngFor="let _ of pageOptions; let i = index;" class="bx--select-option" [value]="i + 1">{{i + 1}}</option>
```

Comment:
Is `let _ of pageOptions` needed?

Reply:
yep, it's part of the nfForOf syntax .... otherwise we won't actually iterate anything

Reply:
Can we name it `page` or something instead of `_`

Reply:
Sure ... my only concern with that is that someone might expect it to contain a valid value